src/srkSmtlib2Defs.ml

```diff
@@ -0,0 +1,43 @@
+type constant =
+  | Int of ZZ.t
+  | Real of QQ.t
+  | String of string
+
+type symbol = string
+
+type index =
+  | Num of ZZ.t
+  | Sym of symbol
+
+type identifier = symbol * (index list)
```

Comment:
This is a surprising type.  Can you explain in a comment?

Reply:
This is the type of identifiers in Smtlib2. See http://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.6-r2017-07-18.pdf bottom of page 96. I'll add a comment in the code as well. An identifier is either a name (symbol) or a name followed by some indexes (e.g. "Vector Int 4" for a 4D vector of integers).